classes/navigationMenu/NavigationMenuDAO.inc.php

```diff
@@ -87,6 +87,23 @@ function getByArea($contextId, $areaName) {
 	}
 
 	/**
+	 * Retrieve all navigation menus by navigation menu area.
+	 * @param $contextId int Context Id
+	 * @param $areaName string Template Area name
+	 * @return NavigationMenu
+	 */
+	function getAllByArea($contextId, $areaName) {
+		$params = array($areaName);
+		$params[] = (int) $contextId;
+		$result = $this->retrieve(
+			'SELECT * FROM navigation_menus WHERE area_name = ? and context_id = ?',
+			$params
+		);
+
+		return new DAOResultFactory($result, $this, '_fromRow');
+	}
+
+	/**
```

Comment:
I don't think you need a new method in the DAO. You should be able to use `getByArea()` for the validation.

Reply:
OK